libs/clj-icat-direct/src/clj_icat_direct/icat.clj

```diff
@@ -38,6 +39,19 @@
   [query & args]
   (k/exec-raw icat [query args] :results))
 
+(defn- run-transaction
+  "Runs the set of passed-in query strings+args in a transaction, returning
+  what the last one returns. This is intended to be used to create temporary
+  tables to circumvent the fact postgresql is bad at doing estimates on CTEs."
+  [& queries]
+  (db/transaction
+    (let [side-effects (drop-last 1 queries)
+          final-query-and-args (last queries)]
+      (doseq [q side-effects]
+        ; uses jdbc/execute! because this can run DDL statements like CREATE TEMPORARY TABLE
```

Comment:
That's probably the issue.
`run-query-string` calls `(k/exec-raw icat [query args] :results)`.
Maybe using `(k/exec-raw q)` here would prevent that exception.